src/auth.ts

```diff
@@ -0,0 +1,45 @@
+/*
+auth.ts - Provides auth related methods for the Bluebutton class
+*/
+
+import BlueButton from "./index";
+import {
+  generateCodeChallenge,
+  generateRandomState,
+} from "./utils/generatePKCE";
+
+export function setCallBackUrl(this: BlueButton, callBackUrl: string): any {
+  this.callBackUrl = callBackUrl;
+}
+
+export function setUsePKCE(this: BlueButton, usePKCE: boolean): any {
+  this.usePKCE = usePKCE;
+}
+
+export function generateAuthorizeUrl(this: BlueButton): any {
+  const BB2_AUTH_URL = this.baseUrl + "/" + this.version + "/o/authorize";
+
+  let pkceParams = "";
+  this.state = generateRandomState();
+
+  if (this.usePKCE) {
+    this.codeChallenge = generateCodeChallenge();
+    pkceParams =
```

Comment:
Not sure why I did it like this in the sample app but... we should really use string templates for this, so much cleaner!
```suggestion
    pkceParams = `&code_challenge_method=S256&code_challenge=${this.codeChallenge.codeChallenge}`
```
The syntax is just using the backticks at the start and end. Inside the `${}` you can use variables, operations, etc. as long as the final result in a string or able to be parsed into a string. You can also add new lines for readability wherever you want.